smt/applications/ego.py

```diff
@@ -298,13 +298,22 @@ def _setup_optimizer(self, fun):
                 random_state=self.options["random_state"],
                 output_in_folded_space=work_in_folded_space,
             )
+            self._sampling_optim = self.mixint.build_sampling_method(
```

Comment:
Is `self._sampling_optim` actually needed when we already have `self._sampling` ? We might also need to set the `random_state` to get reproducibility.

Reply:
- We want to have truly random points to avoid being stuck in the same 20 neighbourhoods again and again. 

- We want to have truly random points so that when we are stuck at an iteration, we are not stuck at the next one.

That is why we do not want seeded starting points for optimization.  

On the other hand, the initial DoE is generating only once, so it makes sense to have a seed for it.

Reply:
Yeah, actually I was guessing such explanation. This is a bug of the current implementation when random_state is set.

The fact is that the current implementation of LHS yields the same sampling again and again when random_state is set.

A better behaviour for sampling method would be to generate different samplings when random_state is set, but it would yield the same sequence of samplings. I will open an issue for this. 

Actually, if we increase `n_start` like you did, it is a way to workaround the problem without introducing `self.sampling_optim` and loosing the reproducibility. What do you think?

Reply:
Yes, I got your point. 
Increasing `n_start` was not a good solution, to be honest. The problem is that we want different starting points at every iteration. 
If we want to have 50 points added by EGO, incresing `n_start` 50 times do not seems reasonable. Even more if we we also increase this number everytime the optimization failed. 
We would just end up with an optimization process that takes hours...

Reply:
Ok I got it. Let's merge!